test/header/CMakeLists.txt

```diff
@@ -82,6 +82,15 @@ TEST(${header_test_name_safe}) {}")
 #endif")
                 endif ()
 
+                # seqan3/std/* must not include platform.hpp (and therefore any other seqan3 header)
+                # See https://github.com/seqan/product_backlog/issues/135
+                if (header MATCHES "seqan3/std/")
+                    file (APPEND "${header_target_source}" "
+#ifdef SEQAN3_DOXYGEN_ONLY
```

Comment:
I don't exactly know how this works in detecting seqan3 headers in /std 
but directly above, when checking platform.hpp it is 
`#ifndef SEQAN3_DOXYGEN_ONLY` (if`n`def)
and here it is 
`#ifdef SEQAN3_DOXYGEN_ONLY` (ifdef)

I guess its right since tests are passing but just double checking :)

Reply:
`platform.hpp` defines `SEQAN3_DOXYGEN_ONLY` and is included by all seqan3 headers.

If we include a seqan3 header, we want `SEQAN3_DOXYGEN_ONLY` to be defined, hence we error if it is not (`ifndef`).

If we include something from `seqan3/std`, we don't want any seqan3 header, i.e. no `SEQAN3_DOXYGEN_ONLY` and hence we error if it is defined (`ifdef`).

Reply:
ah thanks for the explanation :)